integration/cdi/src/main/java/org/apache/cxf/cdi/CdiClassUnwrapper.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.cdi;
+
+import java.util.regex.Pattern;
+
+import org.apache.cxf.common.util.ClassUnwrapper;
+
+class CdiClassUnwrapper implements ClassUnwrapper {
+    private static final Pattern PROXY_PATTERN = Pattern.compile(".+\\$\\$.+Proxy");
```

Comment:
@reta @rmannibucau ok, this uses a pattern based on the discussion.  Do we still want to move outside the CDI module?

Reply:
Thanks @johnament, I don't think it should be moved outside CDI integration, @rmannibucau any particular reasons to do that?

Reply:
Yep the reason is to ensure it is used by all integrations. Very few server uses cdi extension to comply to spec but all use proxies. Since the perf and behavior shouldnt hurt it can be added to the default IMO

Reply:
Hi guys, but the problem is only showing itself on the CDI path ?

Reply:
Not really. it has nothing to do with CDI actually except CDI uses proxies. It will happen each time a lib is using subclassing proxies, CDI is one big consumer but there are a lot of other ways to get it even in plain standalone.

Reply:
Possibly :-) but Spring users are not opening any issues...

Reply:
Oh, indeed, support for Spring is there, you are right; I'm just not sure the proposed ClassUnwrapper should be called there by default every time ClassHelper is called, may be do it in 2 stages, 1) test as part of the CDI 2) move it to the core ?

Reply:
Works for me

Reply:
Sounds like a good plan, @sberyozkin are we good to merge?

Reply:
It's not clear what the plan is, just leave it?

Reply:
Hi Andriy, if all 3 of you are happy, then yes, please merge :-), I did not really contribute myself

Reply:
@johnament Sorry, was not clear, I was referring to @sberyozkin suggestion for 2 stages, let us start from CDI first, and than make a decision if it worth going to core or not. Make sense?